### PR TITLE
Don't run maven update job if we don't need to

### PR DESCRIPTION
**Pull request by Baltoli**
This ports in the equivalent change from the K repo; we don't need to be spending lots of time updating the maven dependencies unless we need to.

### CODE REVIEW

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -13,15 +13,21 @@ jobs:
     name: 'Nix: Maven'
     runs-on: ubuntu-latest
     steps:
-      - name: 'Check out code'
+      - name: 'Check out code, set up Git'
         uses: actions/checkout@v2.3.4
```
Comment (F-WRunTime) [inline .github/workflows/update.yml:17]: Lets bump this v3 for consistency

**Comment by F-WRunTime** [comment]
You mention in the header you want to run this less frequently the trigger however runs on every pull request is that still the frequency you want to see after this PR?

**Comment by Baltoli** [comment]
> every pull request is that still the frequency you want to see after this PR?

Yes, it should run on every PR but only do the costly Maven+Nix work if something will have changed (i.e. a pom.xml or a nix file). This is the same logic as K; the projects have a similar structure wrt. this code

**Comment by Baltoli** [comment]
Ping @F-WRunTime

**Review by F-WRunTime — APPROVED** [review]

**Pull request merged**